### PR TITLE
Added poetry

### LINKED ISSUE #13: Setup Python env
poetry + deps
ruff/black for formatting

### PR DESCRIPTION
**Pull request by DawnOfHell**
added poetry env with base packages for project.

### CODE REVIEW

File: backend/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+[tool.poetry]
+name = "backend"
+version = "0.1.0"
+description = ""
+authors = ["Dawn Heller <shahar.heller3@gmail.com>"]
```
Comment (yuval9313) [inline backend/pyproject.toml]: Where me?

File: .gitignore
Comment (yuval9313) [inline .gitignore:1]: [Check this out](https://github.com/github/gitignore/blob/main/Python.gitignore)

**Review by yuval9313 — CHANGES_REQUESTED** [review]

File: backend/pyproject.toml (hunk 1 above)
Comment (DawnOfHell) [reply backend/pyproject.toml]: you in js frontend boy

Comment (DawnOfHell) [reply backend/pyproject.toml]: jk I just dont know your email, what is it?

Comment (DawnOfHell) [reply backend/pyproject.toml]: done

File: backend/Earthfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
+VERSION 0.7
+FROM python:3.11
+WORKDIR /backend
+
+deps:
+    RUN pip install poetry
+    COPY pyproject.toml ./
+    COPY poetry.lock ./
+    RUN poetry build
+
```
Comment (yuval9313) [inline backend/Earthfile:10]: Missing stages:
docker (for docker builds)
check-format (for formatting)

Comment (DawnOfHell) [reply backend/Earthfile:10]: we didin't decide on a formatter yet tho

File: .gitignore
Comment (yuval9313) [inline .gitignore:1]: Should be scoped to your project

File: backend/Earthfile [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    SAVE ARTIFACT backend /backend
+    SAVE ARTIFACT dist /dist
+
+docker:
+    COPY +build/backend
+    ENTRYPOINT ["python3", ".backend/"]
+    SAVE IMAGE ongkagu-backend:latest
+
+formatter:
+    COPY backend backend
```
Comment (yuval9313) [inline backend/Earthfile]: You already in backend

File: backend/Earthfile [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    RUN pip install poetry
+    RUN poetry install
```
Comment (yuval9313) [inline backend/Earthfile]: You could use +deps

Comment (yuval9313) [reply backend/Earthfile]: Also, due to dir structure I think pyproject.toml would not be available

**Review by yuval9313 — CHANGES_REQUESTED** [review]

File: .gitignore
Comment (DawnOfHell) [reply .gitignore:1]: done

File: backend/Earthfile [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    RUN poetry build
+
+build:
+    FROM +deps
+    COPY backend backend
+    COPY dist dist
+    SAVE ARTIFACT backend /backend
+    SAVE ARTIFACT dist /dist
+
+docker:
```
Comment (yuval9313) [inline backend/Earthfile:18]: Missing 
ARG tag='latest'

**Review by yuval9313 — APPROVED** [review]

**Pull request merged**